### PR TITLE
Add waitUntil method to QtBot

### PR DESCRIPTION
**Pull request by nicoddemus**
As we discussed during the sprint, adding a new `qtbot.waitUntil` method.

### CODE REVIEW

File: pytestqt/qtbot.py [hunk 1]
```diff
@@ -373,6 +375,73 @@ def assertNotEmitted(self, signal):
...
+                # 'assert' form
+                if result is None:
+                    return
+
+                # 'True/False' form
+                if result:
+                    return
+                else:
+                    assert not timed_out(), 'waitUntil timed out in %s miliseconds' % timeout
+            self.wait(10)
```
Comment (The-Compiler) [inline pytestqt/qtbot.py]: Maybe this should be an argument?

Comment (nicoddemus) [inline pytestqt/qtbot.py]: Thought about it, but didn't think it would end up being necessary... perhaps we should wait until someone asks for it?

Comment (The-Compiler) [inline pytestqt/qtbot.py]: Sounds like a plan :wink:

**Comment by nicoddemus** [comment]
Merge? :grin:

**Comment by The-Compiler** [comment]
Yep! :+1:

**Pull request merged**